db/external_sst_file_ingestion_job.cc

```diff
@@ -316,21 +317,9 @@ Status ExternalSstFileIngestionJob::Prepare(
     }
   }
 
-  // TODO: The following is duplicated with Cleanup().
   if (!status.ok()) {
-    IOOptions io_opts;
     // We failed, remove all files that we copied into the db
-    for (IngestedFileInfo& f : files_to_ingest_) {
-      if (f.internal_file_path.empty()) {
-        continue;
-      }
-      Status s = fs_->DeleteFile(f.internal_file_path, io_opts, nullptr);
-      if (!s.ok()) {
-        ROCKS_LOG_WARN(db_options_.info_log,
-                       "AddFile() clean up for file %s failed : %s",
-                       f.internal_file_path.c_str(), s.ToString().c_str());
-      }
-    }
+    DeleteInternalFiles();
```

Comment:
It seems we call Cleanup() after Prepare() return non-ok status: https://github.com/facebook/rocksdb/blob/6e88126dd33219727021d9d9f84b42a47cc647b1/db/db_impl/db_impl.cc#L5850

Reply:
Yes, do you mean that we should remove this `DeleteInternalFiles` call from `Prepare`, so we don't get the "clean up for file failed" for this scenario?

Reply:
Yeah, I did not think about the warnings but just that we don't need to call `DeleteInternalFiles()` here.